### PR TITLE
Add constable server

### PR DESCRIPTION
**Pull request by anstkras**

### CODE REVIEW

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.github.ajalt.clikt.parameters.types.int
+import java.sql.DriverManager
+import java.sql.SQLException
+import java.util.logging.Level
+import java.util.logging.Logger
+
+private val logger: Logger = Logger.getLogger(ConstableServer::class.java.name)
+
+class ConstableServer : CliktCommand() {
+    private val url by option(help = "url").default("127.0.0.1")
```
Comment (dbarashev) [inline com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: I suggest using relatively standard names of postgres connection options which consist of "pg" prefix and host/user/database/port/password suffix: `--pg-host`, `--pg-database`, etc.

With these names the semantics of options are kinda unclear. What might mean `./constable-server --url 10.240.0.10 --name foo` ? Run server on host 10.240.0.10 with the name foo?

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private val name by option(help = "data base name").default("postgres")
+    private val user by option(help = "user").default("postgres")
+    private val password by option(help = "password").default("")
+
+    override fun run() {
+        val connectionUrl = "jdbc:postgresql://$url:$port/$name"
+        try {
+            DriverManager.getConnection(connectionUrl, user, password).use { connection ->
+                connection.createStatement().use { statement ->
+                    statement.executeQuery("SELECT VERSION()").use { resultSet ->
```
Comment (dbarashev) [inline com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: ResultSet is closable too!

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.logging.Logger
+
+private val logger: Logger = Logger.getLogger(ConstableServer::class.java.name)
+
+class ConstableServer : CliktCommand() {
+    private val url by option(help = "url").default("127.0.0.1")
+    private val port by option(help = "port").int().default(5432).validate { require(it in 1024..65535) }
+    private val name by option(help = "data base name").default("postgres")
+    private val user by option(help = "user").default("postgres")
+    private val password by option(help = "password").default("")
```
Comment (dbarashev) [inline com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: Help strings which duplicate option name aren't that useful.

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.validate
+import com.github.ajalt.clikt.parameters.types.int
+import java.sql.DriverManager
+import java.sql.SQLException
+import java.util.logging.Level
+import java.util.logging.Logger
+
+private val logger: Logger = Logger.getLogger(ConstableServer::class.java.name)
```
Comment (dbarashev) [inline com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: I believe that SLF4J + logback is currently the mainstream logging API in JVM world.There's also optional Kotlin wrapper for it (less boilerplate code): https://github.com/MicroUtils/kotlin-logging

File: com.bardsoftware.constable/server/build.gradle [hunk 5]
```diff
@@ -0,0 +1,36 @@
+plugins {
+    id 'java'
```
Comment (dbarashev) [inline com.bardsoftware.constable/server/build.gradle]: Is it really required?

File: com.bardsoftware.constable/server/build.gradle [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    id 'org.jetbrains.kotlin.jvm' version '1.3.50'
```
[the remark this answers is not in the document]
Comment (anstkras) [reply com.bardsoftware.constable/server/build.gradle]: Done

File: com.bardsoftware.constable/server/build.gradle [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+sourceCompatibility = 11
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation("com.github.ajalt:clikt:2.3.0")
+
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
```
[the remark this answers is not in the document]
Comment (anstkras) [reply com.bardsoftware.constable/server/build.gradle]: Done

File: com.bardsoftware.constable/server/build.gradle [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    compile group: 'postgresql', name: 'postgresql', version: '9.1-901-1.jdbc4'
```
[the remark this answers is not in the document]
Comment (anstkras) [reply com.bardsoftware.constable/server/build.gradle]: Done

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt (hunk 3 above)
Comment (anstkras) [reply com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: Done

File: com.bardsoftware.constable/server/build.gradle (hunk 5 above)
Comment (anstkras) [reply com.bardsoftware.constable/server/build.gradle]: Done

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt (hunk 4 above)
Comment (anstkras) [reply com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: Done

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt (hunk 2 above)
Comment (anstkras) [reply com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: It's used inside `use` block.

File: com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+                    statement.executeQuery("SELECT VERSION()").use { resultSet ->
+                        if (resultSet.next()) {
+                            println(resultSet.getString(1))
+                        }
+                    }
+                }
+            }
+        } catch (e: SQLException) {
+            logger.error(e) { "Something went wrong. ${e.message}" }
+            println("Something went wrong. ${e.message}")
```
Comment (dbarashev) [inline com.bardsoftware.constable/server/src/main/kotlin/com/bardsoftware/constable/server/ConstableServer.kt]: You probably don't need println when you have logger call.

**Pull request closed without merging**